src/coreclr/vm/appdomainnative.cpp

```diff
@@ -112,48 +112,36 @@ FCIMPL0(Object*, AppDomainNative::GetLoadedAssemblies)
 } // AppDomainNative::GetAssemblies
 FCIMPLEND
 
-FCIMPL1(Object*, AppDomainNative::IsStringInterned, StringObject* pStringUNSAFE)
+extern "C" void QCALLTYPE String_IsInterned(QCall::StringHandleOnStack str)
 {
-    FCALL_CONTRACT;
-
-    STRINGREF       refString   = ObjectToSTRINGREF(pStringUNSAFE);
-    STRINGREF*      prefRetVal  = NULL;
-
-    HELPER_METHOD_FRAME_BEGIN_RET_1(refString);
-
-    if (refString == NULL)
-        COMPlusThrow(kArgumentNullException, W("ArgumentNull_String"));
+    QCALL_CONTRACT;
 
-    prefRetVal = GetAppDomain()->IsStringInterned(&refString);
+    BEGIN_QCALL;
 
-    HELPER_METHOD_FRAME_END();
+    GCX_COOP();
 
-    if (prefRetVal == NULL)
-        return NULL;
+    STRINGREF refString = str.Get();
+    GCPROTECT_BEGIN(refString);
+    STRINGREF* prefRetVal = GetAppDomain()->IsStringInterned(&refString);
```

Comment:
I wonder if string interning can be moved into managed LoaderAllocator.

Reply:
The string interning is fairly complicated and perf sensitive: https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/stringliteralmap.cpp#L15-L42. I think it would be difficult to get something that has perf parity with the current implementation. It is possible in theory.